### PR TITLE
fix(pubsub): create root async batch sink spans

### PR DESCRIPTION
**Pull request by alevenberg**
Closes https://github.com/googleapis/google-cloud-cpp/issues/12958

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12975)

### CODE REVIEW

**Comment by alevenberg** [comment]
I'm not sure this is the best way to test it. Lmk what you think.

File: google/cloud/pubsub/internal/tracing_message_batch_test.cc [hunk 1]
```diff
@@ -127,6 +128,7 @@ TEST(TracingMessageBatch, Flush) {
   auto mock = std::make_unique<pubsub_testing::MockMessageBatch>();
   EXPECT_CALL(*mock, Flush).WillOnce([] {
     EXPECT_TRUE(ThereIsAnActiveSpan());
+    EXPECT_TRUE(OTelContextCaptured());
     return [](auto) {};
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_message_batch_test.cc:132]: I think you can test that the context is detached by: `EXPECT_FALSE(OTelContextCaptured())` in this lambda.

File: google/cloud/pubsub/internal/tracing_message_batch.cc [hunk 2]
```diff
@@ -135,13 +135,14 @@ std::function<void(future<void>)> TracingMessageBatch::Flush() {
...
   auto batch_sink_spans = MakeBatchSinkSpans(std::move(message_spans));
 
-  // Set the batch sink parent span as the active span. This will always be the
-  // first span in the vector.
-  auto async_scope = internal::GetTracer(internal::CurrentOptions())
-                         ->WithActiveSpan(batch_sink_spans.front());
-
-  return [next = child_->Flush(),
+  // Give the batch sink parent span its own context. This means the span is
+  // root span with no parent. The parent span always be the first span in the
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/tracing_message_batch.cc:139]: "The first span in the vector is the parent of the other spans in the vector." ?

**Review by dbolduc — APPROVED** [review]

File: google/cloud/pubsub/internal/tracing_message_batch_test.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_message_batch_test.cc]: Done

File: google/cloud/pubsub/internal/tracing_message_batch.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_message_batch.cc]: Done

**Pull request merged**